File: Python and JavaScript/Stack/10799_쇠막대기.py
input_data = input()
parenthesis_cnt = len(input_data)       # 괄호 개수
ans = 0                                 # 정답
if parenthesis_cnt != 0:
    stick = 0
    i = 0
    while i <= parenthesis_cnt-1:
        if (input_data[i] == '(' and input_data[i+1] == ')'):       # '()'인 경우 막대기 자르기
            ans += stick                                            # 정답 + 현재 막대 개수
            i += 2                                                  # 인덱스 + 2
            continue
        if input_data[i] == '(':                                    # '('인 경우 정답과 막대 개수 + 1
            ans += 1
            stick += 1
        elif input_data[i] == ')':
           stick -= 1                                               # ')'인 경우 막대 개수 - 1  
        i += 1                                                      # 인덱스 + 1
print(ans)

File: JavaScript/Stack/10799_쇠막대기.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on("line", (line) => {
  const parenthesisLength = line.length;
  let ans = 0;
  let stickCnt = 0;
  let idx = 0;
  while (idx <= parenthesisLength - 1) {
    if (line[idx] === "(" && line[idx + 1] === ")") {
      ans += stickCnt;
      idx += 2;
      continue;
    }
    if (line[idx] === "(") {
      ans += 1;
      stickCnt += 1;
    } else if (line[idx] === ")") {
      stickCnt -= 1;
    }
    idx += 1;
  }
  console.log(ans);
  rl.close();
});

rl.on("close", () => {
  process.exit();
});